virttest/qemu_devices/qcontainer.py

```diff
@@ -2904,17 +2904,25 @@ def dimm_device_define_by_params(self, params, name):
         """
         Create pc-dimm device from params.
         """
-        params = params.object_params("dimm")
+        # Ensure get accurate params for dimm device, avoid irrelevant params
+        # be used by mistake.
+        dimm_params = Params()
+        suffix = '_dimm'
+        for key in list(params.keys()):
+            if key.endswith(suffix):
+                new_key = key.rsplit(suffix)[0]
+                dimm_params[new_key] = params[key]
+
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
-        attrs = qdevices.Dimm.__attributes__[:]
-        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
+        attrs = qdevices.Dimm.__attributes__[dimm_type][:]
+        dev = qdevices.Dimm(params=dimm_params.copy_from_keys(attrs),
                             dimm_type=dimm_type)
         dev.set_param("id", "%s-%s" % ("dimm", name))
-        if dimm_type == "nvdimm" and params.get("nvdimm_uuid"):
+        nvdimm_uuid = dimm_params.get("nvdimm_uuid")
+        if 'uuid' in attrs and nvdimm_uuid:
```

Comment:
Actually, when i replace `dimm_type == 'nvdimm'` with `'uuid' in attrs` here,  i am thinking replace `nvdimm_uuid_dimm` with `uuid_dimm` together.   Use the suffix `_dimm` to limit the namespace and drop prefix `nvdimm`.
It is different with what we discussed offline before that use prefix limit the namespace.

If new dimm type be supported in the future and support `uuid` attribute too,  `nvdimm_uuid_dimm` is not suitable then.
What do you think about this ? @PaulYuuu @luckyh

Reply:
Agree, `uuid_dimm` would be better for future extension reason.

Reply:
Update the patch to use `uuid_dimm`, thanks!

Reply:
+1